### PR TITLE
Fix FIO issues

### PR DESCRIPTION
**Pull request by RazTamir**
Fixes: https://github.com/red-hat-storage/ocs-ci/issues/1276

Signed-off-by: ratamir <ratamir@redhat.com>

### CODE REVIEW

**Comment by RazTamir** [comment]
Verified locally several times

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -110,14 +110,14 @@ def get_fio_results(self):
...
-            if self.fio_thread.running():
-                for sample in TimeoutSampler(
-                    timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.done
-                ):
-                    if sample:
-                        return yaml.safe_load(self.fio_thread.result())
-            if self.fio_thread and self.fio_thread.done():
-                return yaml.safe_load(self.fio_thread.result())
+            for sample in TimeoutSampler(
+                timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.result
```
Comment (shylesh) [inline ocs_ci/ocs/resources/pod.py]: @RazTamir ,
According to https://docs.python.org/3/library/concurrent.futures.html , for result() function "If timeout is not specified or None, there is no limit to the wait time." so Technically result() never returns in our case unless job is Done.

Is that what expected here ? Correct me if I am missing something.

Comment (RazTamir) [reply ocs_ci/ocs/resources/pod.py]: That's true but we have the TimeoutSampler for taking care of it

Comment (shylesh) [reply ocs_ci/ocs/resources/pod.py]: According to my understanding , this line https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/utils.py#L732 will not yield until call is finished.

Am I right ?

Comment (RazTamir) [reply ocs_ci/ocs/resources/pod.py]: You probably do.
Let me run some tests and see. 
I will add timeout in case that's true

**Comment by RazTamir** [comment]
Tested multiple times locally - all passed

**Comment by vasukulkarni** [comment]
@RazTamir are you running jenkins test pr as well?

**Comment by RazTamir** [comment]
> @RazTamir are you running jenkins test pr as well?

yes

**Comment by vasukulkarni** [comment]
CI is failing due to tox issue: ocs_ci/ocs/resources/pod.py:25:1: F401 'ocs_ci.utility.utils.TimeoutSampler' imported but unused

**Comment by RazTamir** [comment]
Not sure why Jenkins didn't report execution result but it passed - https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/3805/console

**Pull request merged**